client/client.go

```diff
@@ -318,6 +286,45 @@ func (c Client) CreateAccessToken(state string) (oauth2.Token, error) {
 	return token, err
 }
 
+func (c Client) do(req *http.Request) (*http.Response, error) {
```

Comment:
Moved this into a single `do` method to avoid the duplication.

Reply:
Largely like this 👍